File: Problem_03/solution_3.lua
-- Project Euler Problem 3
-- Task: The prime factors of 13195 are 5, 7, 13 and 29.
-- What is the largest prime factor of the number 600851475143?

function isPrime(number)
	for i=2, math.pow(number, 0.5) do
		if (number % i == 0) then
			return false
		end
	end
	return true
end

print("Program started...")
bigNumber = 600851475143 -- Starting value
largestFactor = 0

for i=2, bigNumber/2 do
	if isPrime(i) then
		if (bigNumber % i == 0) then
			if (i > largestFactor) then
				print(i)
				largestFactor = i
			end
		end
	end
end

print("The largest prime factor of " .. bigNumber .. " is " .. largestFactor)
print("This programm took " .. os.clock() .. " seconds (" .. os.clock()/60 .. " minutes) for execution.")

-- Solution: 6857 (correct)
-- Script found the solution after a few seconds but checked numbers long time after that... (Overall runtime: more than 5h...)

File: Problem_03/solution_3.py
# Project Euler Problem 3
# Source: https://projecteuler.net/problem=3

import time
start_time = time.time()

def isPrime(number):
	if number % 2 == 0 and number >= 2:
		return False
	for i in range(3, int(number ** 0.5) + 1, 2):
		if number % i == 0:
			return False
	return True

largestFactor, starting_number = 0, 600851475143

for i in range(2, round(starting_number/3)):
	if (starting_number % i == 0) and isPrime(i):
		largestFactor = i

print(i)

print("The script took %s seconds for execution" % (time.time() - start_time))

# Solution: 6857 (Correct)
# Runtime: 